docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,111 @@
+the same way you can pass multiple references with /reference or multiple analyzers
+with /analyzer. There is no short form of the argument. Files passed with relative
+file names are resolved relative to the current working directory the same
+as source files.
+
+Examples:
+
+```
+    csc.exe /editorconfig:.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:..\.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:Subdirectory\.editorconfig
+```
+
+It is expected that the compiler is given the .editorconfig files with casing that
+matches the underlying casing of the file system. The compiler *MUST NOT* do any case correction. So
+
+```
+    csc.exe /editorconfig:Subdirectory\.editorconfig /editorconfig:SubDirectory\.editorconfig
+```
+
+is a valid input to the compiler. The compiler *MAY* choose to issue a warning if two paths given
+are exactly equal by ordinal in a case-sensitive way.
+
+If multiple .editorconfigs apply to a given file (the precise definition of
+"apply" is defined later), and more than one file is defining the same value,
+.editorconfig file listed first in the command arguments takes precedence over
+later files.
+
+# Discovery of .editorconfig files in MSBuild
+
+Discovery of .editorconfig files is not done by csc.exe, but rather the host.
+For csproj projects, a new build task, DiscoverEditorConfigFiles is introduced
+which runs immediately before the CoreCompile target so all inputs are found.
+The inputs to the task are:
+
+- InputFiles: a list of ITaskItems which are the input files from which we must locate .editorconfig
+  files in parent directories. The expected inputs to this include the Compile and AdditionalFile items.
+- RootDirectories: an optional list of root directories which the build task
+  *MUST NOT* ascend higher during any walking. This is a list because a build
+  environment that segregates intermediate directories or build directories
+  outside the root of the source tree might wish to ensure that no walking is
+  ever done outside the source path, or intermediate/output directories, none of which may be a
+  parent of any other root (or indeed on Windows might not even live on the same drive!)
+
+The outputs of this task are:
+
+- EditorConfigFiles: a list of ITaskItems which are paths to discovered .editorconfig files. These *MAY*
+  be relative paths to the current working directory, in the same way that input files *MAY* be relative paths.
+
+The default target maps these items to an EditorConfig item group, that is consumed by the CoreCompile target.
+
+A property DiscoverEditorConfigFiles *MUST* be provided that has the following values it can be set to:
+
+- `SearchToRoot`: the task is ran, and RootDirectories is not populated in any way.
+- `SearchToProject`: the task is ran, and RootDirectories is populated to the project directory.
+- `Off`: the task is not ran at all, via a Condition.
```

Comment:
Where do we discuss the defaults for all of these properties?  This is an area I'm particularly concerned about.